src/include/stir/numerics/FastErf.h

```diff
@@ -0,0 +1,103 @@
+
+/*!
+  \file
+  \ingroup FastErf
+  \brief Implementation of an erf interpolation
+
+  \author Robert Twyman
+*/
+/*
+    Copyright (C) 2022, UCL
+    This file is part of STIR.
+    SPDX-License-Identifier: Apache-2.0
+    See STIR/LICENSE.txt for details
+*/
+
+#include "stir/numerics/BSplines1DRegularGrid.h"
+
+#ifndef __stir_numerics_FastErf__H__
+#define __stir_numerics_FastErf__H__
+
+START_NAMESPACE_STIR
+
+
+/*! \ingroup numerics
+   \name FastErf
+   \brief The class acts as a potentially faster way to compute many erf values by precomputing the function at
+   regularly spaced intervals. [BSplines, linear, nearest neighbour] interpolation methods are available.
+   Note, nearest neighbour is fastest and BSplines slowest method.
+*/
+class FastErf
+{
+private:
+
+  //! The number of erf samples to take from -\cmaximum_sample_value to \cmaximum_sample_value
+  int _num_samples = 1000;
+
+  //! The sampling period, computed as \cmaximum_sample_value / \c_num_samples)
+  double _sampling_period;
+
+//  //! Used to check if setup has been run before parameter changes
+//  bool _is_setup = false;
+
+  //! BSplines object using linear interpolation
+  BSpline::BSplines1DRegularGrid<double, double> spline;
+
+  /*! The upper bound value x value of erf(x) used in sampling. Default erf(x=5) ~= 1.
+   * The negative \cmaximum_sample_value is used as the lower bound.
+   */
+  double maximum_sample_value = 5;
+
+  //! a vector/list of stored erf values
+  std::vector<double> erf_values_vec;
+
+public:
+  FastErf(){}
```

Comment:
I find it a bit cleaner to provide a default value to the `explicit` function below. Less code, and now the 1000 is hidden in the `private` member. You could leave it, but then needs to be documented (we normally run doxygen ignoring private members)

Reply:
Let me see if I understand this right. `int _num_samples = 1000;` won't be documented (by default) because it is a private member variable. You want me to add/modify the constructor to:

```
FastErf()
{
  _num_samples = 1000;
}

explicit FastErf(int num_samples)
{
  this->_num_samples = num_samples;
}
```

Reply:
Or 
```
FastErf()
{
  FastErf(1000);
}

explicit FastErf(int num_samples)
{
  this->_num_samples = num_samples;
}
```

Reply:
neither of the above is self-documented.
```c++
explicit FastErf(int num_samples = 1000)
{
  this->_num_samples = num_samples;
}
```
would do the trick.

Reply:
In which case, should the:
```
FastErf()
{
  FastErf(1000);
}
```
 be removed and then use:

```
  explicit FastErf(const int num_samples = 1000, const float maximum_sample_value = 5)
  {
    this->_num_samples = num_samples;
    this->maximum_sample_value = maximum_sample_value;
  }
```